lib/core/utils/separated_list.dart

```diff
@@ -0,0 +1,12 @@
+extension SpratedList<T> on List<T> {
```

Comment:
- Add documentation 
- remove generic and use Widget
- move to UI util
- Optional: use Divider as const separator